catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/forms/model/FilterNodeAssertionSupport.java

```diff
@@ -59,8 +73,180 @@ static void assertTemplatedNode(
+     * Begin data validation on a non-terminal elements.
+     *
+     * @param expressionMapper mapping operation for data structure traversal.
+     * @return assertion object used to take the next validation step.
+     */
+    public JAXBCollectionAssertion withExpression(
+        Function<? super T, List<JAXBElement<?>>> expressionMapper) {
+      notNull(expressionMapper);
+      return new JAXBCollectionAssertion(expressionMapper.apply(elementValue));
+    }
+
+    /**
+     * Begin validation on terminal elements whose values conform to {@code List<JAXBElement<?>>}.
+     *
+     * @param expressionMapper mapping operation for data structure traversal.
+     * @return assertion object used to take the next validation step.
+     */
+    public JAXBPropertyValueAssertion verifyExpression(
+        Function<? super T, List<JAXBElement<?>>> expressionMapper) {
+      return new JAXBPropertyValueAssertion(
+          (p, v) -> assertExpression(expressionMapper.apply(elementValue), p, v));
+    }
+
+    /**
+     * Begin validation on terminal elements whose values conform to {@code List<Object>}.
+     *
+     * @param expressionOrAnyMapper mapping operation for data structure traversal.
+     * @return assertion object used to take the next validation step.
+     */
+    public JAXBPropertyValueAssertion verifyExpressionOrAny(
+        Function<? super T, List<Object>> expressionOrAnyMapper) {
+      return new JAXBPropertyValueAssertion(
+          (p, v) -> assertExpressionOrAny(expressionOrAnyMapper.apply(elementValue), p, v));
+    }
+  }
+
+  static class JAXBPropertyValueAssertion {
+    private final BiConsumer<String, Serializable> assertion;
+
+    private JAXBPropertyValueAssertion(BiConsumer<String, Serializable> assertion) {
+      this.assertion = assertion;
+    }
+
+    /**
+     * Specify the data to test for. This is a terminal operation.
+     *
+     * @param property the expected value for the property field, also called the {@code
+     *     <ValueReference/>}.
+     * @param value the expected value for the value field, also called the {@code <Literal/>}.
+     */
+    public void withData(String property, Serializable value) {
+      assertion.accept(property, value);
+    }
+  }
+
+  static class JAXBCollectionAssertion {
+    private final List<JAXBElement<?>> elements;
+
+    private JAXBCollectionAssertion(List<JAXBElement<?>> elements) {
+      this.elements = elements;
+    }
+
+    /**
+     * Provide a set of assertions that define the collection of {@link JAXBElement}s.
+     *
+     * <p>The number of assertions provided should be equal to the number of {@link JAXBElement}s
+     * present in the list or the test will fail. The first assertion will be executed against the
+     * first {@link JAXBElement}, the second assertion against the 2nd element, and so on in this
+     * fashion.
+     *
+     * @param assertions set of assertions to map to the {@link JAXBElement}s.
+     */
+    @SafeVarargs
```

Comment:
❗️ Need to verify this is OK. 
https://docs.oracle.com/javase/tutorial/java/generics/nonReifiableVarargsType.html#heap_pollution

Related SO post. 
https://stackoverflow.com/questions/12462079/potential-heap-pollution-via-varargs-parameter